File: src/HookBridge.Domain/Entities/AuditLog.cs
namespace HookBridge.Domain.Entities;

/// <summary>
/// Represents an immutable admin audit trail entry.
/// </summary>
public sealed class AuditLog : BaseEntity
{
    public string TenantId { get; set; } = string.Empty;

    public string? UserId { get; set; }

    public string? UserEmail { get; set; }

    public string Action { get; set; } = string.Empty;

    public string ResourceType { get; set; } = string.Empty;

    public string? ResourceId { get; set; }

    public string? Description { get; set; }

    public object? Metadata { get; set; }

    public string? IpAddress { get; set; }

    public string? UserAgent { get; set; }

}

File: src/HookBridge.Dashboard/src/types/auditLog.ts
import type { PagedRequest } from './pagination';

export type AuditLogResponse = {
  id: string;
  tenantId: string;
  userId: string | null;
  userEmail: string | null;
  action: string;
  resourceType: string;
  resourceId: string | null;
  description: string | null;
  metadata: unknown;
  ipAddress: string | null;
  userAgent: string | null;
  createdAt: string;
};

export type AuditLogSearchRequest = PagedRequest & {
  userId?: string;
  userEmail?: string;
  action?: string;
  resourceType?: string;
  resourceId?: string;
  fromDate?: string;
  toDate?: string;
};
